src/Domain/Creators/Cylinder.cpp

```diff
@@ -34,104 +24,113 @@ Cylinder::Cylinder(
-          Wedge3DPrism{Wedge2D{inner_radius_, outer_radius_, 0.0, 1.0,
-                               OrientationMap<2>{std::array<Direction<2>, 2>{
-                                   {Direction<2>::upper_xi(),
-                                    Direction<2>::upper_eta()}}},
-                               use_equiangular_map_},
-                       Affine{-1.0, 1.0, lower_bound_, upper_bound_}},
-          Wedge3DPrism{Wedge2D{inner_radius_, outer_radius_, 0.0, 1.0,
-                               OrientationMap<2>{std::array<Direction<2>, 2>{
-                                   {Direction<2>::lower_eta(),
-                                    Direction<2>::upper_xi()}}},
-                               use_equiangular_map_},
-                       Affine{-1.0, 1.0, lower_bound_, upper_bound_}},
-          Wedge3DPrism{Wedge2D{inner_radius_, outer_radius_, 0.0, 1.0,
-                               OrientationMap<2>{std::array<Direction<2>, 2>{
-                                   {Direction<2>::lower_xi(),
-                                    Direction<2>::lower_eta()}}},
-                               use_equiangular_map_},
-                       Affine{-1.0, 1.0, lower_bound_, upper_bound_}},
-          Wedge3DPrism{Wedge2D{inner_radius_, outer_radius_, 0.0, 1.0,
-                               OrientationMap<2>{std::array<Direction<2>, 2>{
-                                   {Direction<2>::upper_eta(),
-                                    Direction<2>::lower_xi()}}},
-                               use_equiangular_map_},
-                       Affine{-1.0, 1.0, lower_bound_, upper_bound_}});
-
-  if (use_equiangular_map_) {
-    coord_maps.emplace_back(
-        make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
-            Equiangular3DPrism{
-                Equiangular(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
-                            inner_radius_ / sqrt(2.0)),
-                Equiangular(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
-                            inner_radius_ / sqrt(2.0)),
-                Affine{-1.0, 1.0, lower_bound_, upper_bound_}}));
-  } else {
-    coord_maps.emplace_back(
-        make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
-            Affine3D{Affine(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
-                            inner_radius_ / sqrt(2.0)),
-                     Affine(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
-                            inner_radius_ / sqrt(2.0)),
-                     Affine{-1.0, 1.0, lower_bound_, upper_bound_}}));
+  const size_t number_of_shells = 1 + radial_partitioning_.size();
+  const size_t number_of_discs = 1 + height_partitioning_.size();
+  std::vector<PairOfFaces> pair_of_faces{};
+  if (is_periodic_in_z_) {
+    const size_t corners_per_layer = 4 * (number_of_shells + 1);
+    const size_t num_corners = number_of_discs * corners_per_layer;
+    const PairOfFaces center{
```

Comment:
You can remove the `const` and instead `std::move` these objects into the `push_back`s.